src/lib/utils.ts

```diff
@@ -429,3 +435,58 @@ export function hashData(
 
   return hashFunction.method(data);
 }
+
+/**
+ * Hashes the data received with the specified hashing function,
+ * and compares the result with the provided hash.
+ *
+ * @throws *Error* in case of a mismatch of the hashes.
+ */
+export function hashAndCompare(
+  data,
+  hash: string,
+  lowerCaseHashFunction: SUPPORTED_HASH_FUNCTIONS,
+) {
+  const jsonHash = hashData(data, lowerCaseHashFunction);
+
+  // throw error if hash mismatch
+  if (jsonHash !== hash) {
+    throw new Error(`
+              Hash mismatch, returned JSON ("${jsonHash}") is different than the one
+              linked from the ERC725Y smart-contract: "${hash}"
```

Comment:
I think this part of the comment "ERC725Y smart-contract" is outside of the scope of this function.

Let's just keep "returned JSON hash {xxx} is different from expected hash {yyy}"

Because this function may be called in another context.

This message with "ERC725Y smart-contract" may make more sense in the function: `getDataFromExternalSources` or in the `fetchData` one, because there, we know the hash comes from the smart contract